src/main/java/org/zalando/failsafeactuator/config/FailsafeInjectionConfiguration.java

```diff
@@ -10,38 +10,43 @@
  */
 package org.zalando.failsafeactuator.config;
 
-import net.jodah.failsafe.CircuitBreaker;
+import java.lang.annotation.Annotation;
 
 import org.springframework.beans.factory.InjectionPoint;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
 import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Scope;
-import org.zalando.failsafeactuator.service.CircuitBreakerRegistry;
+import org.zalando.failsafeactuator.service.CircuitBreakerFactory;
 import org.zalando.failsafeactuator.service.FailsafeBreaker;
 
-import java.lang.annotation.Annotation;
+import net.jodah.failsafe.CircuitBreaker;
 
 /** Configuration class which enables the usage of {@link FailsafeBreaker} annotation. */
 @Configuration
 @Conditional(FailsafeAutoConfiguration.FailsafeCondition.class)
 public class FailsafeInjectionConfiguration {
 
-  @Autowired
-  private CircuitBreakerRegistry circuitBreakerRegistry;
-
+	@Autowired
+	private ApplicationContext applicationContext;
+	
   @Bean
   @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
-  public CircuitBreaker circuitBreaker(InjectionPoint ip) {
+  public CircuitBreaker circuitBreaker(InjectionPoint ip) throws Exception {
     FailsafeBreaker annotation = null;
     for (Annotation a : ip.getAnnotations()) {
       if (a instanceof FailsafeBreaker) {
         annotation = (FailsafeBreaker) a;
         break;
       }
     }
-    return circuitBreakerRegistry.getOrCreate(annotation.value());
+    
+    CircuitBreakerFactory circuitBreakerFactory = new CircuitBreakerFactory();
+    AutowireCapableBeanFactory factory = applicationContext.getAutowireCapableBeanFactory();
```

Comment:
You should use here the factory you created? Furthermore I think we are loosing information about the identifier here?

Reply:
Hi @MALPI, I set the identifier in the next line. ( line 56).

Reply:
Hi @MALPI , how to make the link between the newly created CircuitBreaker and the identifier ? On line 56, this is how I link the 2 together. Is this correct ?

Reply:
No I don't think so. First think you should use the `circuitBreakerFactory`, which you actually should inject.

By this all created Beans are already known to the application context.

I think there still should be a registry in place which checks if a Bean with this identifier is already in place or not. You can check [Put it to rest Registry](https://github.com/zalando-stups/put-it-to-rest/blob/master/src/main/java/org/zalando/putittorest/Registry.java) as reference.

It should happen in general in a way like it is done in [putittorest](https://github.com/zalando-stups/put-it-to-rest/tree/master/src/main/java/org/zalando/putittorest).

Reply:
OK @MALPI I will take a look